src/Appium.Net/Appium/Service/AppiumCommandExecutor.cs

```diff
@@ -12,22 +12,28 @@
 //See the License for the specific language governing permissions and
 //limitations under the License.
 
+using OpenQA.Selenium.Appium.Windows;
 using OpenQA.Selenium.Remote;
 using System;
+using System.Collections.Generic;
 using System.Reflection;
 
 namespace OpenQA.Selenium.Appium.Service
 {
     internal class AppiumCommandExecutor : ICommandExecutor
     {
         private readonly AppiumLocalService Service;
-        private readonly Uri URL;
+        private static Uri URL;
+        private static TimeSpan CommandTimeout;
+
         private ICommandExecutor RealExecutor;
         private bool isDisposed;
-        private const string IdempotencyHeader = "X-Idempotency-Key";
```

Comment:
What was the reason behind removing this feature?

Reply:
Selenium 4 no longer allows us to directly interaction with HttpClient
See:
https://github.com/SeleniumHQ/selenium/blob/trunk/dotnet/src/webdriver/Remote/HttpCommandExecutor.cs

Would love to hear of something (there than reflection) that would work.

Reply:
You can still do this with v4. The move to `HttpClient` broke it, because the `HttpWebRequest` object was no longer used, but we put the ability to modify headers back in [this commit](https://github.com/SeleniumHQ/selenium/commit/c66b1ea6bce7e09219512254b9f0b6bc31acf477)

Reply:
Not sure of the best way to do this.
This looks like the easiest implementation
```
    internal class AppiumCommandExecutor : HttpCommandExecutor
    {
        private readonly AppiumLocalService Service;
        private bool isServiceDisposed;

        /// <summary>
        /// Initializes a new instance of the <see cref="AppiumCommandExecutor"/> class
        /// </summary>
        /// <param name="addressOfRemoteServer">Address of the WebDriver Server</param>
        /// <param name="timeout">The timeout within which the server must respond.</param>
        public AppiumCommandExecutor(Uri addressOfRemoteServer, TimeSpan timeout)
            : base(addressOfRemoteServer, timeout, true)
        {
            Service = null;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="AppiumCommandExecutor"/> class
        /// </summary>
        /// <param name="service">Appium local service</param>
        /// <param name="timeForTheServerResponding">The timeout within which the server must respond.</param>
        internal AppiumCommandExecutor(AppiumLocalService service, TimeSpan timeForTheServerResponding)
            : base(service.ServiceUrl, timeForTheServerResponding, true)
        {
            Service = service;
        }

        protected override void OnSendingRemoteHttpRequest(SendingRemoteHttpRequestEventArgs eventArgs)
        {
            eventArgs.AddHeader("X-Idempotency-Key", Guid.NewGuid().ToString());
            base.OnSendingRemoteHttpRequest(eventArgs);
        }

        protected override void Dispose(bool disposing)
        {
            if (!isServiceDisposed)
            {
                if (disposing)
                {
                    Service?.Dispose();
                }

                isServiceDisposed = true;
            }

            base.Dispose(disposing);
        }
    }
```
and it seems to work, but really terribly confident